sklearn/grid_search.py

```diff
@@ -194,14 +193,13 @@ def __iter__(self):
         else:
             # Always sort the keys of a dictionary, for reproducibility
             items = sorted(self.param_distributions.items())
-            while len(samples) < self.n_iter:
+            for _ in repeat(None, self.n_iter):
```

Comment:
> I didn't want to create a list here (and in Python 2.* range would create a list).

Do you really expect the length of the list to be so important that it matters?

> However I should probably use six.moves.range here.

That would be an option.